### PR TITLE
feat: introduce parser and cli

### PR DESCRIPTION
**Pull request by NektariosFifes**
**Description**

- Basic cli functionality.
- eslint and added related npm packages
- initialized readme
- Input loop to make sure the user provides a correctly formated ,and existent async api File

**Related issue(s)**
#18

### CODE REVIEW

File: Async-Simulator/README.md
Comment (AceTheCreator) [inline Async-Simulator/README.md:5]: Npm library which aim is to making your backend services testable at scale. @NektariosFifes what do you think?

File: Async-Simulator/src/cliUI/LoadingBar.js
Comment (AceTheCreator) [inline Async-Simulator/src/cliUI/LoadingBar.js:14]: Are these spaces necessary or maybe because you still wanna add somethings there later?

**Review by AceTheCreator — CHANGES_REQUESTED** [review]
Looks good from here, great job man 👍. @jonaslagoni you might wanna review this PR

File: Async-Simulator/src/cliUI/LoadingBar.js
Comment (NektariosFifes) [reply Async-Simulator/src/cliUI/LoadingBar.js:14]: they are dots that aim to make it feel like a bar filling. https://www.fileformat.info/info/unicode/char/2591/index.htm

**Comment by jonaslagoni** [comment]
@NektariosFifes whenever the PR is ready to be reviewed you can select reviewers to the right side of this PR which sends a request to me that I should take a look at it. This is great in some cases where last-minute changes are still needed before it is getting reviewed, otherwise you can always create the PR as a draft which marks the PR as in progress.

Is it ready to be reviewed? 😄

**Comment by NektariosFifes** [comment]
> @NektariosFifes whenever the PR is ready to be reviewed you can select reviewers to the right side of this PR which sends a request to me that I should take a look at it. This is great in some cases where last-minute changes are still needed before it is getting reviewed, otherwise you can always create the PR as a draft which marks the PR as in progress.
> 
> Is it ready to be reviewed? 😄

nope

**Comment by NektariosFifes** [comment]
@jonaslagoni  if you want u can review pr

File: Async-Simulator/simulator.sh
Comment (jonaslagoni) [inline Async-Simulator/simulator.sh:3]: Do you still need the bash script?

File: Async-Simulator/src/bin/cli.js
Comment (jonaslagoni) [inline Async-Simulator/src/bin/cli.js:96]: You can unwrap the `run` function since there is no need for that functionality to be wrapped in a function 🙂

File: Async-Simulator/src/bin/dockerCli.js
Comment (jonaslagoni) [inline Async-Simulator/src/bin/dockerCli.js:1]: Why is this file included? 🤔

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
So left quite a few comments in multiple locations, but generally, it looks good 👍 🙂 

A few general comments:
- Why have you created the `Async-Simulator` folder? 🤔 I think we can unwrap this right?
- Any files that are not used at the moment, this includes empty files, UI library files (if they are not used in the CI at the moment, should be removed from this PR. Feel free to save them on another branch 🙂 
- Keep referring to the library as Fluffy Robot, at least until another name is found, this is to ensure consistency when referenced 

Let me know if you have any questions about my comments, and ensure you evaluate my comments and not just blindly accept them as you might have a different opinion or know something I dont 🙂

File: Async-Simulator/src/bin/cli.js
Comment (NektariosFifes) [reply Async-Simulator/src/bin/cli.js]: i just keep that there in case i want to use it in smth other than the cli.

File: Async-Simulator/Dockerfile
Comment (NektariosFifes) [reply Async-Simulator/Dockerfile]: ok

**Comment by NektariosFifes** [comment]
ok i think i fixed them :) @jonaslagoni

File: package.json
Comment (pawanakhil) [inline package.json:39]: eslint dependencies should be in `devDependencies`

**Review by pawanakhil — CHANGES_REQUESTED** [review]
Looks good mate ! Few changes requested, have a look it. :slightly_smiling_face:

File: src/parser/AsyncParser.js
Comment (NektariosFifes) [reply src/parser/AsyncParser.js]: yes i added it thinking it could be exposed to end user for utilization. It is not needed,removed

File: Async-Simulator/Dockerfile
Comment (NektariosFifes) [reply Async-Simulator/Dockerfile]: but it would be better to add in next pr

**Comment by jonaslagoni** [comment]
Our CI system requires you to name the title of the PR something like `feat: simulation lib` for it properly understand it. Maybe a better title would be `feat: introduce parser and cli`?

File: src/parser/AsyncParser.js
Comment (jonaslagoni) [inline src/parser/AsyncParser.js:41]: What's the reason for wrapping the actual parsing in another function? 🤔

File: src/parser/index.js
Comment (jonaslagoni) [inline src/parser/index.js:2]: I think we might need to rename the parser to something like `scenarioParser` as otherwise people will get confused with github.com/asyncapi/parser-js

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Got some more comments 👍 

Almost there 🎉

File: src/bin/cli.js
Comment (NektariosFifes) [reply src/bin/cli.js:85]: I am doing it for abstraction. i wanted these function to act as interfaces. i might add more 'interfaces' to verifyInput_ParseFile that need to be called before returning parsed data like adding some things to the  handlingContenxt.ParsedAndFormated.

Comment (NektariosFifes) [reply src/bin/cli.js:85]: also returning th ParsedAndFormated data is optional as it will be handled internally

File: src/parser/AsyncParser.js
Comment (NektariosFifes) [reply src/parser/AsyncParser.js:41]: i want to able to divide the parsing process in different parts. In case of some error i want it to be handled and caught in the cli and not handled in the parser. I think that because i want all the logging to be done by one interface so that i am able to add smth on top of it in the future.(add a stream to persistently log in a file etc)

Comment (NektariosFifes) [reply src/parser/AsyncParser.js]: @jonaslagoni  w8 dont merge it yet one last thing

Comment (NektariosFifes) [reply src/parser/AsyncParser.js]: ok i removed it for simplicity

File: src/bin/cli.js
Comment (NektariosFifes) [reply src/bin/cli.js]: i removed it for simplicity.

File: package.json
Comment (jonaslagoni) [inline package.json:36]: Unused dependency 🙂

File: package.json
Comment (jonaslagoni) [inline package.json:38]: Unused dependency 🙂

File: package.json
Comment (jonaslagoni) [inline package.json:41]: Unused dependency

File: src/bin/cli.js
Comment (jonaslagoni) [inline src/bin/cli.js:14]: Mismatch return type

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Almost there 🎉 

Only a final touchup, please go through all the dependencies and remove any that are unused 🙂

File: src/bin/cli.js
Comment (NektariosFifes) [reply src/bin/cli.js]: @jonaslagoni  ok

File: package.json
Comment (jonaslagoni) [inline package.json:39]: What are you using these two dependencies for? 🤔

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Make sure you update `package-lock.json` and commit it after the last dependencies have been assessed 🙂

File: package.json
Comment (NektariosFifes) [reply package.json:39]: i think they installed because i used npm install with another node version.

Comment (NektariosFifes) [reply package.json:39]: i guess it is for using a specific npm version installed in node_modules

Comment (NektariosFifes) [reply package.json:39]: could these be usefull?

Comment (jonaslagoni) [reply package.json:39]: Only if they are used, otherwise no, anything that is never used should never be committed 🙂

Comment (jonaslagoni) [reply package.json:39]: This is why I am asking whether you use the dependencies anywhere in your code, which would require those who run the library to need them 🙂 (I know they need npm to install this library but to run it is another story)

Comment (NektariosFifes) [reply package.json]: @jonaslagoni  ok

**Comment by jonaslagoni** [comment]
@NektariosFifes you have two lint errors that have to be dealt with 🙂 The security lint warning can be disabled I think 🤔

File: src/index.js
Comment (jonaslagoni) [inline src/index.js:1]:
```suggestion
exports.scenarioParser = require('./parser').scenarioParser;
```

**Review by jonaslagoni — COMMENTED** [review]
Last suggestion and lets merge 👍

File: src/index.js
Comment (NektariosFifes) [reply src/index.js]: there is smth i always forget about xD

**Review by jonaslagoni — APPROVED** [review]
LGTM, great job! 👍

**Pull request merged**
